raiden-ts/src/services/utils.ts

```diff
@@ -119,22 +119,24 @@ export async function pfsInfo(
 
     const start = Date.now();
     const res = await fetch(url + '/api/v1/info', { mode: 'cors' });
+    const rtt = Date.now() - start;
     const text = await res.text();
     assert(res.ok, [ErrorCodes.PFS_ERROR_RESPONSE, { text }]);
     const info = decode(PathInfo, jsonParse(text));
 
-    assert(info.price_info.lte(pfsMaxFee), [
-      ErrorCodes.PFS_TOO_EXPENSIVE,
-      { price: info.price_info.toString() },
-    ]);
+    const { payment_address: address, price_info: price } = info;
+    assert(price.lte(pfsMaxFee), [ErrorCodes.PFS_TOO_EXPENSIVE, { price }]);
     pfsAddressCache_.set(url, Promise.resolve(info.payment_address));
+    const validTill =
+      (await serviceRegistryContract.callStatic.service_valid_till(address)).toNumber() * 1e3;
```

Comment:
This does an RPC call to the ETH-node.
I don't fully understand what this code does:

https://github.com/raiden-network/light-client/blob/b51dff44e492e757992ce72171ef76cab7621984/raiden-ts/src/services/epics/helpers.ts#L211
Will the `pfsInfo` function execute on every path-request and thus make a JSON-RPC call to the eth node every time?
If not I have a follow up question:
- Do you update the `validTill` at some point like you do with the auto-discovered PFSes?
Because if the PFS calls `deposit` again to prolong the registration this would not be updated.

Reply:
This calls RPC every time a transfer is about to be made, because we also need to fetch up-to-date PFS info (in case they updated the price, etc), so there's no caching here.

Reply:
Yes, for the PFS-HTTP endpoint it makes sense - but technically you could easily spare at least the ethnode RPC-call here and rather update the validTill from emitted blockchain events upon `deposit` calls. This is just an optimisation - I'm just mentioning this since we saw quite some RPC calls to the ethnode in the python client that could be avoided and the ethnode is one of the bottlenecks of concern.

But in any case, please go ahead with merging this when you consider not worth it (right now).

Reply:
Good call. We happen to actually already monitor and persist the mapping of valid registered services in `state.services` (monitored from the event), so it was actually just a matter of checking if the service address is in the dict and fetching from the static contract call just as a fallback. Thanks for the headsup, we really need to look on those unneeded calls more closely.

Reply:
Done :heavy_check_mark: